lib/ansible/modules/windows/win_certificate_info.ps1

```diff
@@ -0,0 +1,145 @@
+        {
+            $cert_info.intended_purposes = $extension.EnhancedKeyUsages.FriendlyName -as [string[]]
+        }
+        elseif ($extension -is [System.Security.Cryptography.X509Certificates.X509KeyUsageExtension])
+        {
+            $cert_info.key_usages = $extension.KeyUsages.ToString().Split(',').Trim() -as [string[]]
+        }
+        elseif ($extension -is [System.Security.Cryptography.X509Certificates.X509SubjectKeyIdentifierExtension])
+        {
+            $cert_info.ski = $extension.SubjectKeyIdentifier
+        }
+        elseif ($extension.Oid.value -eq '2.5.29.17')
+        {
+            $sans = $extension.Format($true).Split("`r`n", [System.StringSplitOptions]::RemoveEmptyEntries)
+            foreach ($san in $sans)
+            {
+                $san_parts = $san.Split("=")
+                if ($san_parts.Length -ge 2 -and $san_parts[0].Trim() -eq 'DNS Name')
+                {
+                    $cert_info.dns_names += $san_parts[1].Trim()
+                }
+            }
+        }
+    }
+    return $cert_info
+}
+
+$store_location_values = ([System.Security.Cryptography.X509Certificates.StoreLocation]).GetEnumValues() | ForEach-Object { $_.ToString() }
+
+$spec = @{
+    options = @{
+        thumbprint     = @{ type = "str"; required = $false }
+        store_name     = @{ type = "str"; default = "My"; }
+        store_location = @{ type = "str"; default = "LocalMachine"; choices = $store_location_values; }
+    }
+}
+
+$module = [Ansible.Basic.AnsibleModule]::Create($args, $spec)
+
+$thumbprint = $module.Params.thumbprint
+$store_name = $module.Params.store_name
+$store_location = [System.Security.Cryptography.X509Certificates.Storelocation]"$($module.Params.store_location)"
+
+$store = New-Object -TypeName System.Security.Cryptography.X509Certificates.X509Store -ArgumentList $store_name, $store_location
+try
+{
+    $store.Open([System.Security.Cryptography.X509Certificates.OpenFlags]::ReadWrite)
+}
+catch [System.Security.Cryptography.CryptographicException]
+{
+    $module.FailJson("Unable to open the store as it is not readable: $($_.Exception.Message)", $_)
+}
+catch [System.Security.SecurityException]
+{
+    $module.FailJson("Unable to open the store with the current permissions: $($_.Exception.Message)", $_)
+}
+catch
+{
+    $module.FailJson("Unable to open the store: $($_.Exception.Message)", $_)
+}
+
+$module.Result.exists = $false
+$module.Result.certificates = @()
+
+try
+{
+    if ($null -ne $thumbprint)
+    {
+        $found_certs = $store.Certificates.Find([System.Security.Cryptography.X509Certificates.X509FindType]::FindByThumbprint, $thumbprint, $false)
+    }
+    else
+    {
+        $found_certs = $store.Certificates
+    }
+
+    if ($found_certs.Count -gt 0)
+    {
+        $module.Result.exists = $true
+        [array]$module.Result.certificates = $found_certs | ForEach-Object { Get-CertificateInfo -cert $_ }
```

Comment:
I wonder if it's a good idea to make sure the output is sorted and not rely on whatever algorithm .NET has. If we wanted to it would just be a matter of adding `| Sort-Object -Property {$_.<key>}`

What are your thoughts?

Reply:
I like the idea of sorting so that it is a predictable order, but I don't know which property makes the most sense to sort by. I think the default in the certificate store GUI is to sort by the "issued to" property.